src/MIDebugEngine/Engine.Impl/DebuggedThread.cs

```diff
@@ -368,9 +368,19 @@ private bool TryGetTidFromTargetId(string targetId, out uint tid)
                 // In gdb coredumps the thread name is in the form:" LWP <thread-id>"
                 return true;
             }
+            else if (targetId.StartsWith("pid ", StringComparison.OrdinalIgnoreCase))
+            {
+                // In QNX gdb coredumps the thread name is in the form:"pid <pid> tid <pid>", eg "pid 4194373 tid 308"
+                int tid_pos = targetId.IndexOf("tid ", StringComparison.Ordinal);
+                int len = targetId.Length - (tid_pos + 4);
+                if (len > 0 && System.UInt32.TryParse(targetId.Substring(tid_pos + 4, len), out tid) && tid != 0)
```

Comment:
nit: Use a constant for `tid ` and use '<const_name>.Length` instead of '4' so it is obvious what the '4' means.

Example:
```
const string tidPrefix = "tid ";
int tid_pos = targetId.IndexOf(tidPrefix, ...)
int len = targetId.Lenth - (tid_pos + tidPrefix.Length);
if (... && UInt32.TryParse(targetId.Substring(tid_pos + tidPrefix.Length...)
```